client/state/login/reducer.js

```diff
@@ -38,10 +42,11 @@ export const requestSuccess = createReducer( null, {
 
 export const twoFactorAuth = createReducer( null, {
 	[ LOGIN_REQUEST ]: () => null,
-	[ LOGIN_REQUEST_SUCCESS ]: ( state, { data } ) => data || null,
+	[ LOGIN_REQUEST_SUCCESS ]: ( state, { data, rememberMe } ) => data ? { ...data, remember_me: rememberMe } : null,
 	[ TWO_FACTOR_AUTHENTICATION_LOGIN_REQUEST_FAILURE ]: ( state, { twoStepNonce } ) => Object.assign( {}, state, {
 		two_step_nonce: twoStepNonce
 	} ),
+	[ LOGIN_TWOFACTOR_UPDATE_NONCE ]: ( state, { twoStepNonce } ) => ( { ...state, two_step_nonce: twoStepNonce } ),
```

Comment:
Another options is we can dispatch both actions [here](https://github.com/Automattic/wp-calypso/blob/0810616558d96e8285849164a4c9a13306abb436/client/state/login/actions.js#L138-L142):
```js
dispatch( {
	type: TWO_FACTOR_AUTHENTICATION_LOGIN_REQUEST_FAILURE,
	error: errorMessage,
} );
dispatch ( {
       type: LOGIN_TWOFACTOR_UPDATE_NONCE,
	twoStepNonce: get( error, 'response.body.data.two_step_nonce' ),
} );
```